### PR TITLE
fix(Core/Spell): Instant Statue

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
- Currently, the "pedestal" creature is not being removed if you remove the aura.
- One of the auras that Instant Statue Pedestal applies on the player is not removed (fixed with spell_linked_spell)

## Tests Performed:
- Builds
- Tested in-game

## How to Test the Changes:

1. .add 54212
2. use the item
3. Rightclick the aura to remove it
4. See that the pedestal is removed

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by pklloveyou** [comment]
Tested

File: src/server/scripts/Spells/spell_item.cpp [hunk 1]
```diff
@@ -1546,6 +1546,56 @@ class spell_item_summon_argent_knight :  public SpellScriptLoader
...
+    {
+        PrepareAuraScript(spell_item_instant_stature_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_INSTANT_STATUE))
+            {
+                return false;
+            }
+            return true;
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_item.cpp:1570]: could be one line, for example:

```cpp
return static_cast<bool>(sSpellMgr->GetSpellInfo(SPELL_INSTANT_STATUE));
```

or

```cpp
return !!sSpellMgr->GetSpellInfo(SPELL_INSTANT_STATUE);
```

**Comment by FrancescoBorzi** [comment]
@Kitzunu if you have double-checked it after the last change, for me it's good to go

**Comment by Kitzunu** [comment]
Yep I did. Works =}

**Comment by FrancescoBorzi** [comment]
perfect! thanks for your contribution @Kitzunu

**Comment by Kitzunu** [comment]
Changed it to ValidateSpellInfo so it can't be merged before https://github.com/azerothcore/azerothcore-wotlk/pull/4323

File: data/sql/updates/pending_db_world/rev_1611180684314340400.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1611180684314340400');
+
+DELETE FROM `spell_script_names` WHERE `spell_id` = 75731;
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
```
Comment (BarbzYHOOL) [inline data/sql/updates/pending_db_world/rev_1611180684314340400.sql:4]: in that case, you could have used "replace into" as it affect only one row, although it's only working in mysql based databases if i recall right

(not required to edit it, was just for info)

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1611180684314340400.sql:4]: yeye, I you're right.
But personally I, and I know other members also, prefers DEL/INS

**Comment by pklloveyou** [comment]
Tested

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**